skpro/distributions/halfnormal.py

```diff
@@ -0,0 +1,76 @@
+# copyright: skpro developers, BSD-3-Clause License (see LICENSE file)
+"""Half-Normal probability distribution."""
+
+__author__ = ["SaiRevanth25"]
+
+import pandas as pd
+from scipy.stats import halfnorm, rv_continuous
+
+from skpro.distributions.adapters.scipy import _ScipyAdapter
+
+
+class HalfNormal(_ScipyAdapter):
+    r"""Half-Normal distribution.
+
+    This distribution is univariate, without correlation between dimensions
+    for the array-valued case.
+
+    The pdf is defined only for \( x \geq 0 \). For \( x < 0 \), \( f(x) = 0 \).
+    The distribution is "cut off" at \( x = 0 \). There is no mass assigned to
+    negative values; they are entirely excluded from the distribution.
```

Comment:
```suggestion
    The pdf is defined only for :math:`\( x \geq 0 \)`. For :math:`\( x < 0 \), \( f(x) = 0 \)`.
    The distribution is `cut off` at :math:`\( x = 0 \)`. There is no mass assigned to
    negative values; they are entirely excluded from the distribution.
```

otherwise you have
<img width="708" alt="Screenshot 2024-05-26 at 22 01 53" src="https://github.com/sktime/skpro/assets/115561717/a6c99e9e-0f95-419c-9e09-33227fb9035b">

Reply:
I think the notation is unusual, I would also shorten this a lot. I would just append $, x>0; \mbox{otherwise} 0$ to the formula.

Reply:
Updated it!! please let me know if anything else is to be done! Thank you.